test/e2e/user_workload_monitoring_test.go

```diff
@@ -724,6 +724,46 @@ func assertTenancyForMetrics(t *testing.T) {
 			t.Fatalf("failed to query Thanos querier: %v", err)
 		}
 	}
+
+	// Check that the account doesn't have to access the rules endpoint.
+	err = framework.Poll(5*time.Second, time.Minute, func() error {
+		// The tenancy port (9092) is only exposed in-cluster so we need to use
+		// port forwarding to access kube-rbac-proxy.
+		host, cleanUp, err := f.ForwardPort(t, "thanos-querier", 9092)
+		if err != nil {
+			t.Fatal(err)
+		}
+		defer cleanUp()
+
+		client := framework.NewPrometheusClient(
+			host,
+			token,
+			&framework.QueryParameterInjector{
+				Name:  "namespace",
+				Value: userWorkloadTestNs,
+			},
+		)
+
+		resp, err := client.Do("GET", "/api/v1/rules", nil)
+		if err != nil {
+			return err
+		}
+		defer resp.Body.Close()
+
+		b, err := ioutil.ReadAll(resp.Body)
+		if err != nil {
+			return err
+		}
+
+		if resp.StatusCode/100 == 2 {
+			return fmt.Errorf("unexpected status code response, got %d (%s)", resp.StatusCode, framework.ClampMax(b))
```

Comment:
nit: can we be a bit more explicit about what we expect? i.e. `expected request to be rejected, but got status code response %d (%s)` ?

Reply:
:+1: